helix-term/src/commands/typed.rs

```diff
@@ -1445,12 +1449,13 @@ pub(super) fn goto_line_number(
             let text = doc.text().slice(..);
             let line = doc.selection(view.id).primary().cursor_line(text);
             cx.editor.last_line_number.get_or_insert(line + 1);
+
+            let line = args[0].parse::<usize>()?;
+            goto_line_without_jumplist(cx.editor, NonZeroUsize::new(line));
+            let (view, doc) = current!(cx.editor);
```

Comment:
This line can be removed: view and doc are already in scope

Reply:
I juggled some pieces around here to address your comments, but the borrow rules here make it tricky. Any feedback is appreciated.

Reply:
You usage is correct `cx.editor` is passed `goto_line_without_jumplist` mutably so any references into `cx.editor` (like `view` and `doc`) are invalidated and need to be recreated. This pattern is commonly used in other commands so you did the right thing

Reply:
> I juggled some pieces around here to address your comments, but the borrow rules here make it tricky. Any feedback is appreciated.

Sorry, I should have noticed that or tried it and the borrow checker would have complained to me instead.